features/steps/thens/attributes.py

```diff
@@ -134,7 +134,7 @@ def step_impl(context, inst, constraint, num):
     inst = str(inst)
     op = misc.stmt_to_op(constraint)
     if not op(len(inst), num):
-        yield ValidationOutcome(inst=inst, expected=num, observed=len(inst), severity=OutcomeSeverity.ERROR) 
+        yield ValidationOutcome(inst=inst, expected={'length':num, 'expected_or_observed':'expected'}, observed={'length': len(inst), 'expected_or_observed':'observed', 'inst':inst}, severity=OutcomeSeverity.ERROR) 
```

Comment:
There has to be a better way of determining `expected` vs `observed` on the front end.  That being said we can leave as-is for now and clean up in a future refactor.